### ISSUE TITLE
[BUG]: Incorrect check for cudax flag

### ISSUE THREAD

**Issue by osayamenja** [NVIDIA/cccl#2423]
### Is this a duplicate?

- [x] I confirmed there appear to be no [duplicate issues](https://github.com/NVIDIA/cccl/issues) for this bug and that I agree to the [Code of Conduct](CODE_OF_CONDUCT.md)

### Type of Bug

Something else

### Component

Not sure

### Describe the bug

In [cccl-config.cmake](https://github.com/NVIDIA/cccl/blob/main/lib/cmake/cccl/cccl-config.cmake), the check for `CCCL_ENABLE_UNSTABLE` always returns false.

Instead, the check should be
```CMake
if(DEFINED ENV{CCCL_ENABLE_UNSTABLE} AND "$ENV{CCCL_ENABLE_UNSTABLE}")
     # ....
endif()
```

### How to Reproduce

Set `CCCL_ENABLE_UNSTABLE=ON` or any truth value and try to import `CCCL::cudax`

### Expected behavior

Doing the above should include the `cudax` target.

### Operating System

N/A

### nvidia-smi output

N/A

### NVCC version

N/A

**Comment by jrhemstad** [comment]
@alliepiper any chance you already took care of this as part of the cleanup you're doing?

**Comment by alliepiper** [comment]
Nope, but I'll add it to the list.

**Comment by alliepiper** [comment]
This is working as intended. That is not an environment variable, but rather a cmake variable that should be defined before `find_package(CCCL)`.

```cmake
set(CCCL_ENABLE_UNSTABLE ON)
find_package(CCCL)
```

**Comment by osayamenja** [comment]
> This is working as intended. That is not an environment variable, but rather a cmake variable that should be defined before `find_package(CCCL)`.
> 
> set(CCCL_ENABLE_UNSTABLE ON)
> find_package(CCCL)

Yes, verified that this works! I misunderstood the [README](https://github.com/NVIDIA/cccl/tree/main/cudax), thinking it was an env variable.
